widget/button_test.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package widget_test
+
+import (
+	"image"
+	"testing"
+
+	"gioui.org/io/key"
+	"gioui.org/io/router"
+	"gioui.org/io/system"
+	"gioui.org/layout"
+	"gioui.org/op"
+	"gioui.org/widget"
+)
+
+func TestButtonKeypresses(t *testing.T) {
+	var (
+		ops op.Ops
+		r   router.Router
+		b1  widget.Clickable
+		b2  widget.Clickable
+	)
+	gtx := layout.NewContext(&ops, system.FrameEvent{Queue: &r})
+	layout := func() {
+		layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+			layout.Rigid(func(gtx layout.Context) layout.Dimensions {
+				return b1.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+					return layout.Dimensions{Size: image.Pt(100, 100)}
+				})
+			}),
+			layout.Rigid(func(gtx layout.Context) layout.Dimensions {
+				return b2.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+					return layout.Dimensions{Size: image.Pt(100, 100)}
+				})
+			}),
+		)
+	}

```

Comment:
If you only need to offset `b2`, `layout` can be written without Flex, by using `op.Offset` just before laying out `b2`:

```go
defer op.Offset(image.Pt(100, 100)).Push(gtx.Ops).Pop()
```